both gnome and kde seems to claim the application.menu name in the XDG menu dir.
We can't have that - or having kde and gnome conflicting would be bad bad bad, so both  parties divert
from the name. The path is hardcoded though, so needs to be patched in.


--- a/kded/kbuildsycoca.cpp
+++ b/kded/kbuildsycoca.cpp
@@ -307,7 +307,7 @@
      connect(g_vfolder, SIGNAL(newService(const QString &, KService::Ptr *)),
              this, SLOT(slotCreateEntry(const QString &, KService::Ptr *)));
 
-     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("applications.menu", true);
+     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("kde4-applications.menu", true);
 
      KServiceGroup::Ptr entry = g_bsgf->addNew("/", kdeMenu->directoryFile, KServiceGroup::Ptr(), false);
      entry->setLayoutInfo(kdeMenu->layoutList);
--- a/kded/CMakeLists.txt
+++ b/kded/CMakeLists.txt
@@ -67,7 +67,7 @@
 
 ########### install files ###############
 
-install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus )
+install( FILES applications.menu  DESTINATION ${SYSCONF_INSTALL_DIR}/xdg/menus RENAME kde4-applications.menu)
 install( FILES kdedmodule.desktop DESTINATION  ${SERVICETYPES_INSTALL_DIR} )
 install( FILES kded.upd           DESTINATION  ${DATA_INSTALL_DIR}/kconf_update )
 
